spatialmath/box.go

```diff
@@ -145,47 +145,47 @@ func (b *box) ToProtobuf() *commonpb.Geometry {
 }
 
 // CollidesWith checks if the given box collides with the given geometry and returns true if it does.
-func (b *box) CollidesWith(g Geometry) (bool, error) {
+func (b *box) CollidesWith(g Geometry, collisionBufferMM float64) (bool, error) {
 	if other, ok := g.(*box); ok {
-		return boxVsBoxCollision(b, other), nil
+		return boxVsBoxCollision(b, other, collisionBufferMM), nil
 	}
 	if other, ok := g.(*sphere); ok {
-		return sphereVsBoxCollision(other, b), nil
+		return sphereVsBoxCollision(other, b, collisionBufferMM), nil
 	}
 	if other, ok := g.(*capsule); ok {
-		return capsuleVsBoxCollision(other, b), nil
+		return capsuleVsBoxCollision(other, b, collisionBufferMM), nil
 	}
 	if other, ok := g.(*point); ok {
-		return pointVsBoxCollision(other.position, b), nil
+		return pointVsBoxCollision(other.position, b, collisionBufferMM), nil
 	}
 	return true, newCollisionTypeUnsupportedError(b, g)
 }
 
-func (b *box) DistanceFrom(g Geometry) (float64, error) {
+func (b *box) DistanceFrom(g Geometry, collisionBufferMM float64) (float64, error) {
 	if other, ok := g.(*box); ok {
-		return boxVsBoxDistance(b, other), nil
+		return boxVsBoxDistance(b, other, collisionBufferMM), nil
 	}
 	if other, ok := g.(*sphere); ok {
 		return sphereVsBoxDistance(other, b), nil
 	}
 	if other, ok := g.(*capsule); ok {
-		return capsuleVsBoxDistance(other, b), nil
+		return capsuleVsBoxDistance(other, b, collisionBufferMM), nil
 	}
 	if other, ok := g.(*point); ok {
 		return pointVsBoxDistance(other.position, b), nil
 	}
 	return math.Inf(-1), newCollisionTypeUnsupportedError(b, g)
 }
 
-func (b *box) EncompassedBy(g Geometry) (bool, error) {
+func (b *box) EncompassedBy(g Geometry, collisionBufferMM float64) (bool, error) {
```

Comment:
Same thing here, I dont see why this should need to take in this new parameter.  We should likely change the implementations to no longer need this.